src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonPropertyInfo.cs

```diff
@@ -1178,5 +1178,195 @@ private class MyClassCustomConverterFactory : JsonConverterFactory
+            Assert.Null(value.ExtensionData);
+        }
+
+        [Theory]
+        [InlineData(typeof(IDictionary<string, object>))]
+        [InlineData(typeof(IDictionary<string, JsonElement>))]
+        [InlineData(typeof(Dictionary<string, object>))]
+        [InlineData(typeof(Dictionary<string, JsonElement>))]
+        [InlineData(typeof(ConcurrentDictionary<string, JsonElement>))]
+        [InlineData(typeof(JsonObject))]
+        public static void EnablingExtensionData_SupportedPropertyType(Type type)
+        {
+            var jti = JsonTypeInfo.CreateJsonTypeInfo(typeof(ClassWithExtensionDataAttribute), new());
+            var jpi = jti.CreateJsonPropertyInfo(type, "ExtensionData");
+
+            Assert.False(jpi.IsExtensionData);
+            jpi.IsExtensionData = true;
+            Assert.True(jpi.IsExtensionData);
+        }
+
+        [Theory]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(object))]
+        [InlineData(typeof(ClassWithExtensionDataAttribute))]
+        [InlineData(typeof(List<int>))]
+        [InlineData(typeof(IDictionary<string, string>))]
+        [InlineData(typeof(Dictionary<string, int>))]
+        public static void EnablingExtensionData_UnsupportedPropertyType_ThrowsInvalidOperationException(Type type)
+        {
+            var jti = JsonTypeInfo.CreateJsonTypeInfo(typeof(ClassWithExtensionDataAttribute), new());
+            var jpi = jti.CreateJsonPropertyInfo(type, "ExtensionData");
+
+            Assert.False(jpi.IsExtensionData);
+            Assert.Throws<InvalidOperationException>(() => jpi.IsExtensionData = true);
+        }
+
+        public class ClassWithExtensionDataAttribute
+        {
+            public int Value1 { get; set; }
+            public int Value2 { get; set; }
+
+            [JsonExtensionData]
+            public Dictionary<string, object> ExtensionData { get; set; }
+        }
+
+        [Fact]
+        public static void DefaultJsonTypeInfoResolver_JsonPropertyInfo_ReturnsMemberInfoAsAttributeProvider()
+        {
+            var resolver = new DefaultJsonTypeInfoResolver();
+            JsonTypeInfo jti = resolver.GetTypeInfo(typeof(ClassWithFieldsAndProperties), new());
+
+            Assert.Equal(2, jti.Properties.Count);
+
+            JsonPropertyInfo fieldPropInfo = jti.Properties.First(prop => prop.Name == "Field");
+            FieldInfo fieldInfo = typeof(ClassWithFieldsAndProperties).GetField("Field");
+            Assert.NotNull(fieldInfo);
+            Assert.Same(fieldInfo, fieldPropInfo.AttributeProvider);
+
+            JsonPropertyInfo propertyPropInfo = jti.Properties.First(prop => prop.Name == "Property");
+            PropertyInfo propInfo = typeof(ClassWithFieldsAndProperties).GetProperty("Property");
+            Assert.NotNull(propInfo);
+            Assert.Same(propInfo, propertyPropInfo.AttributeProvider);
```

Comment:
we could loop here and test couple of other classes as well since json name matches property name

Reply:
Not sure I follow?

Reply:
```csharp
Assert.True(ti.Properties.Count > 0);
foreach (var prop in ti.Properties)
{
    PropertyInfo propInfo = typeof(ClassWithFieldsAndProperties).GetProperty(prop.Name);
    Assert.NotNull(propInfo);
    Assert.Same(propInfo, prop.AttributeProvider);
}
```

Reply:
What specific functionality do you feel is not being covered by the existing test?